source/neuropod/multiprocess/mq/ipc_message_queue_impl.hh

```diff
@@ -152,7 +162,32 @@ void IPCMessageQueue<UserPayloadType>::send_message(const WireFormat &msg)
         SPDLOG_TRACE("OPE: Sending IPC control message {}.", msg.type);
     }
 
-    send_queue_->send(&msg, sizeof(msg), 0);
+    // Send w/ timeout + heartbeat check
+    while (true)
```

Comment:
can you explain a bit why there's a while loop? it seems if send does not time out and heartbeat is not lost we'll indefinitely retry?

Reply:
>  if send does not time out and heartbeat is not lost we'll indefinitely retry?

If send does not time out, we'll return (lines 173 and 174).

If send times out, we'll keep retrying until it works or we lose the heartbeat. This is the same behavior as before (except the whole application doesn't terminate when we lose a heartbeat).